bitcoin/src/consensus/mod.rs

```diff
@@ -19,6 +24,52 @@ pub use self::params::Params;
 pub use self::validation::{
     verify_script, verify_script_with_flags, verify_transaction, verify_transaction_with_flags,
 };
+use crate::prelude::Vec;
+
+/// Deserialize any decodable type from a hex string.
+pub fn deserialize_hex<T: Decodable>(hex: &str) -> Result<T, DecodeHexError> {
+    let v = Vec::from_hex(hex)?;
```

Comment:
Before refactoring there was Read on HexIterator to avoid intermediate allocation https://github.com/rust-bitcoin/bitcoin_hashes/pull/135
We may want to do something similar and improve this in a future PR

Reply:
Yes, good point. I can look into that. Thanks

Reply:
Ha, good reviewing @RCasatta, there is an implementation of `io::Read` on `hex-conservative::HexToBytesIter` that we can use.